### PR TITLE
Implement MarshalJSON for query results

### LINKED ISSUE #112: Query unsupported type calculation.NewResultArrayFromValueTypesFn
Pulled this from the logingester logs:
```
02:07:09.546747[I] executing query [{type aggregation} {body {
  "namespace": "rtapi",
  "calculations": [
    {
      "op": "COUNT"
    }
  ],
  "time_range": "24h",
  "time_unit": 1,
  "group_by": [
    "@timestamp"
  ]
}}]
02:07:09.549066[W] error executing query: received HTTP status: 500 Internal Server Error. body: {"state":"Error","error":"json: unsupported type: calculation.NewResultArrayFromValueTypesFn"}
```

### PR DESCRIPTION
**Pull request by xichen2020**
cc @notbdu @black-adder 

This PR implements `MarshalJSON` for the various types of query results.

Fixes #112

### CODE REVIEW

File: calculation/value.go [hunk 1]
```diff
@@ -16,18 +18,40 @@ const (
 	StringType
 )
 
+var (
+	nullBytes = []byte("null")
```
Comment (notbdu) [inline calculation/value.go]: Do we want to define this somewhere in a shared location?

File: query/result_group.go [hunk 2]
```diff
@@ -5,15 +5,28 @@ import (
 	"github.com/xichen2020/eventdb/x/compare"
 )
 
+type nullResultGruop struct {
```
Comment (black-adder) [inline query/result_group.go]: typo

File: query/result_group.go [hunk 3]
```diff
@@ -5,15 +5,28 @@ import (
...
+	Key    *string                 `json:"key"` // This should always be a nil pointer to represent null key
+	Values calculation.ResultArray `json:"values"`
+}
+
+type nullResultGruopsJSON struct {
```
Comment (black-adder) [inline query/result_group.go]: ditto

File: query/single_key_result_gruops_test.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
+package query
```
Comment (black-adder) [inline query/single_key_result_gruops_test.go]: file name typo

File: query/time_bucket_result.go [hunk 5]
```diff
@@ -60,3 +61,30 @@ func (r *TimeBucketResults) MergeInPlace(other *TimeBucketResults) error {
...
+	}
+	res := timeBucketResultsJSON{
+		Granularity: r.BucketSizeNanos,
+		Buckets:     buckets,
+	}
+	return json.Marshal(res)
+}
+
+type timeBucketJSON struct {
+	StartAt int64 `json:"startAt"` // In nanoseconds
```
Comment (black-adder) [inline query/time_bucket_result.go]: startAtNanos?

File: calculation/value.go (hunk 1 above)
Comment (xichen2020) [reply calculation/value.go]: Hm a bit overkill given this is only used in two places in two separate packages.

File: query/time_bucket_result.go (hunk 5 above)
Comment (xichen2020) [reply query/time_bucket_result.go]: Sure.

**Pull request merged**